September 2023 releases (#163)
conda-incubator/conda-dot-org: blog/2023-10-12-september-releases.md

---
title: "September 2023 Releases"
slug: "2023-10-12-september-releases"
description: |
    Conda 23.9.0, conda-build 3.27.0, conda-libmamba-solver 23.9.0/23.9.1, conda-index 0.3.0, and pycosat 0.6.5/0.6.6 have been released! 🎉
authors: [kenodegard]
tags: [announcement, conda, conda-build, conda-libmamba-solver, conda-index, pycosat]
---

The September 2023 releases included updates to five conda community projects: conda, conda-build, conda-libmamba-solver, conda-index, and pycosat! 🎉 Each of these have been released to both `main` and `conda-forge`.

## Changes in Conda [23.9.0](https://github.com/conda/conda/releases/tag/23.9.0)

To update conda, run:
```bash
conda install -n base conda=23.9.0
```

### 📢 Special Announcement 📢

> > Conda's default solver will change to [`conda-libmamba-solver`](https://conda.github.io/conda-libmamba-solver/) in a __special 23.10.0 release__ in the near future!

Author: ```suggestion
:::info

Conda's default solver will change to [`conda-libmamba-solver`](https://conda.github.io/conda-libmamba-solver/) in a __special 23.10.0 release__ in the near future!

:::
```

---
title: "September 2023 Releases"
slug: "2023-10-12-september-releases"
description: |
    Conda 23.9.0, conda-build 3.27.0, conda-libmamba-solver 23.9.0/23.9.1, conda-index 0.3.0, and pycosat 0.6.5/0.6.6 have been released! 🎉
authors: [kenodegard]
tags: [announcement, conda, conda-build, conda-libmamba-solver, conda-index, pycosat]
---

The September 2023 releases included updates to five conda community projects: conda, conda-build, conda-libmamba-solver, conda-index, and pycosat! 🎉 Each of these have been released to both `main` and `conda-forge`.

## Changes in Conda [23.9.0](https://github.com/conda/conda/releases/tag/23.9.0)

To update conda, run:
```bash
conda install -n base conda=23.9.0
```

### 📢 Special Announcement 📢

:::info

Conda's default solver will change to [`conda-libmamba-solver`](https://conda.github.io/conda-libmamba-solver/) in a __special 23.10.0 release__ in the near future!

:::

You can already benefit from it _today_ by [configuring your conda installation to use it](https://conda.github.io/conda-libmamba-solver/getting-started/#usage) (e.g. by running `conda config --set solver libmamba`).

> The current "classic" solver is based on [pycosat](https://github.com/conda/pycosat)/[Picosat](http://fmv.jku.at/picosat/) and will remain part of conda for the forseeable future, a fallback is possible and available.

Reviewer: ```suggestion
The current "classic" solver is based on [pycosat](https://github.com/conda/pycosat)/[Picosat](http://fmv.jku.at/picosat/) and will remain part of conda for the foreseeable future. A fallback is possible and available.
```

> The current "classic" solver is based on [pycosat](https://github.com/conda/pycosat)/[Picosat](http://fmv.jku.at/picosat/) and will remain part of conda for the forseeable future, a fallback is possible and available.

Author: superseded by https://github.com/conda-incubator/conda-dot-org/pull/163#discussion_r1357212349

* Only apply `defaults::pkg` workarounds for the default value `default_channels`.
* Users won't be able to override pinned specs with incompatible CLI specs anymore. Instead they must modify their pinned specs explicitly.

### 📄 What's New in Documentation? 📄

* Document intentional deviations from conda's `classic` solver behavior.

* * *

## Changes in Conda-Index [0.3.0](https://github.com/conda/conda-index/releases/tag/0.3.0)

To update conda-index, run:
```bash
conda install -n base conda-index=0.3.0
```

### ✨ What's New? ✨

* Add `--run-exports` to generate CEP-12 compliant `run_exports.json` documents for each subdir.
* Don't pretty-print `repodata.json` by default, saving time and space.
* Require conda >= 4.14.

* * * 

## Changes in Pycosat [0.6.5](https://github.com/conda/pycosat/releases/tag/0.6.5)/[0.6.6](https://github.com/conda/pycosat/releases/tag/0.6.6)

To update conda-index, run:
```bash
conda install -n base pycosat=0.6.6
```

### ✨ What's New? ✨

* Pycosat 0.6.4 accidentally did not include the changes intended to be released. Pycosat 0.6.5 includes those changes.

> * Use `PyMem_Calloc()` to initialize memory to `0`.

Reviewer 2: ```suggestion
* `PyMem_Calloc()` is utilized in order to initialize memory to `0`.
```

* * *

### We ❤️ Our Community

> Altogether we had 10 (!) new contributors this release cycle; thank you to all of our open source community members for helping making these improvements possible.

Reviewer 2: ```suggestion
Altogether we had 11 (!) new contributors this release cycle; thank you to all of our open source community members for helping making these improvements possible.
```

* [@boldorider4](https://github.com/boldorider4) made their first contribution in [conda-build#4960](https://github.com/conda/conda-build/pull/4960)

> * [@DaveKaretnyk](https://github.com/DaveKaretnyk) made their first contribution in [conda-build#5004](https://github.com/conda/conda-build/pull/5004)
> * [@jmcarpenter2](https://github.com/jmcarpenter2) made their first contribution in [conda#13034](https://github.com/conda/conda/pull/13034)

Reviewer 2: ```suggestion
* [@DaveKaretnyk](https://github.com/DaveKaretnyk) made their first contribution in [conda-build#5004](https://github.com/conda/conda-build/pull/5004)
* [@dholth](https://github.com/dholth) made their first contribution in [pycosat#54](https://github.com/conda/pycosat/pull/54)
* [@jmcarpenter2](https://github.com/jmcarpenter2) made their first contribution in [conda#13034](https://github.com/conda/conda/pull/13034)
```

```bash
conda install -n base conda=23.9.0
```

### 📢 Special Announcement 📢

:::info

Conda's default solver will change to [`conda-libmamba-solver`](https://conda.github.io/conda-libmamba-solver/) in a __special 23.10.0 release__ in the near future!

:::

You can already benefit from `conda-libmamba-solver` _today_ by [configuring your conda installation to use it](https://conda.github.io/conda-libmamba-solver/getting-started/#usage) (e.g. by running `conda config --set solver libmamba`).

The current "classic" solver is based on [pycosat](https://github.com/conda/pycosat)/[Picosat](http://fmv.jku.at/picosat/) and will remain part of conda for the foreseeable future. A fallback is possible and available.

Additional details on this change as well as the full announcement can be found in the [conda 23.9.0 changelog](https://github.com/conda/conda/blob/main/CHANGELOG.md#2390-2023-09-27).

<!-- truncate -->

### ✨ What's New? ✨

* A new "auth handler" plugin hook has been added for conda. See the new [`conda-auth`](https://github.com/conda-incubator/conda-auth) plugin for an example.
* Index cache metadata is locked by default; a `--no-lock` option is also now available.

> * A new`context.register_envs` option can control whether to register environments in `~/.conda/environments.txt` when they are created. Defaults to true.

Reviewer 3: ```suggestion
* A new `context.register_envs` option can control whether to register environments in `~/.conda/environments.txt` when they are created. Defaults to true.
```

* How tasks for `libsolv` are created has been rewritten, making use of `libmamba`'s `add_pin` features (requires `libmambapy >=1.5.1`).
* Name-only pins now lock the corresponding package if installed.
* The `.solv` cache for repodata is used if available and recent.

### 🔧 What Got Fixed? 🔧

* The solver no longer bounces between two compatible solutions when the same command is run twice in a row.
* Commands with no channels passed are handled gracefully.
* Workaround added for missing `noarch` field in returned `PackageRecord` payload.
* A bug where the `--prune` flag was not working correctly in `conda env update` commands has been fixed.
* Environments are no longer aggressively updated to higher priority channels under certain conditions.
* No more injection of channels from installed packages that do not exist or are unavailable.
* All configured channels in `PackagesNotFoundError` exceptions print correctly.
* No more crashes when a `MatchSpec` with a build string is specified in the CLI and there's a pinned spec for the same package name.
* `defaults::pkg` workarounds are only applied for the default value `default_channels`.

> * Users won't be able to override pinned specs with incompatible CLI specs anymore. Instead they must modify their pinned specs explicitly.

Reviewer 3: ```suggestion
* Users won't be able to override pinned specs with incompatible CLI specs anymore. Instead, they must modify their pinned specs explicitly.
```

### 📄 What's New in Documentation? 📄

* Intentional deviations from conda's `classic` solver behavior are now documented.

* * *

## Changes in Conda-Index [0.3.0](https://github.com/conda/conda-index/releases/tag/0.3.0)

To update conda-index, run:
```bash
conda install -n base conda-index=0.3.0
```

### ✨ What's New? ✨

* `--run-exports` has been added to generate CEP-12 compliant `run_exports.json` documents for each `subdir`.
* Pretty-print `repodata.json` is no longer default, saving time and space.
* Version requirement: `conda >= 4.14`.

* * * 

## Changes in Pycosat [0.6.5](https://github.com/conda/pycosat/releases/tag/0.6.5)/[0.6.6](https://github.com/conda/pycosat/releases/tag/0.6.6)

To update conda-index, run:
```bash
conda install -n base pycosat=0.6.6
```

### ✨ What's New? ✨

* Pycosat 0.6.4 accidentally did not include the changes intended to be released. Pycosat 0.6.5 includes those changes.
* `PyMem_Calloc()` is utilized in order to initialize memory to `0`.

* * *

### We ❤️ Our Community

> Altogether we had 11 (!) new contributors this release cycle; thank you to all of our open source community members for helping making these improvements possible.

Reviewer 3: ```suggestion
Altogether, we had 11 (!) new contributors this release cycle; thank you to all of our open source community members for helping making these improvements possible.
```
